src/lib.rs

```diff
@@ -42,9 +42,18 @@
 //!
 
 #![no_std]
-
+#![cfg(not(feature = "std"))]
 use core::ops::{Bound, RangeBounds};
 
+#[cfg(feature = "std")]
+use std::ops::{Bound, RangeBounds};
+
+#[cfg(feature = "std")]
+extern crate std;
```

Comment:
First I did like https://github.com/staticintlucas/stringslice/pull/2#discussion_r1739711120

But then I encountered some problems with tests, so that's how I figured it out. Maybe there's a better solution

Reply:
Ah, yeah, for tests you need `std`.

You should do something like `#[cfg_attr(not(any(feature = "std", test)), no_std)]` to enable `std` for tests